packages/core/services/DatabaseService/index.ts

```diff
@@ -140,18 +141,25 @@ export default abstract class DatabaseService {
     }
 
     public async prepareSourceMetadata(sourceMetadata: Source): Promise<void> {
-        if (sourceMetadata.type && sourceMetadata.uri) {
-            this.deleteDataSource(this.SOURCE_METADATA_TABLE);
-            this.dataSourceToAnnotationsMap.clear();
+        const isOldSource = sourceMetadata.name === this.sourceMetadataName;
```

Comment:
I always find that things with "old" or "new" sort of get lost with time. Might want to add some context here or consider some different names.